src/runner.ts

```diff
@@ -1,17 +1,163 @@
+            reportPath = pt.join(reportPath, "coverage.xml");
+        }
+
+        reportPath = reportPath.replace(/\\/g, "/");
+
+        try {
+            // Use glob to find the matching report paths
+            reportPaths = glob.sync(reportPath);
+        } catch (error) {
+            return undefined;
+        }
+
+        if (!reportPaths) {
+            return undefined;
+        }
+
+        if (reportPaths.length == 1) {
+            core.info(messagesFormatter.format(messages.found_coverage_report, reportPaths[0]));
+        } else if (reportPaths.length > 1) {
+            core.info(messagesFormatter.format(messages.found_multiple_coverage_report, reportPaths.length));
+        }
+
+        return reportPaths;
+    }
+
+    private getJavaFilePath(parasoftToolOrJavaRootPath: string | undefined): string | undefined {
+        let installDir = parasoftToolOrJavaRootPath || process.env.JAVA_HOME;
+
+        if (!installDir || !fs.existsSync(installDir)) {
+            core.warning(messages.java_or_parasoft_tool_install_dir_not_found);
+            return undefined;
+        }
+
+        const javaFilePath = this.doGetJavaFilePath(installDir);
+        if (!javaFilePath) {
+            core.warning(messagesFormatter.format(messages.java_not_found_in_java_or_parasoft_tool_install_dir));
+        } else {
+            core.debug(messagesFormatter.format(messages.found_java_at, javaFilePath));
+        }
+
+        return javaFilePath;
+    }
+
+    private doGetJavaFilePath(installDir: string): string | undefined {
+        core.debug(messagesFormatter.format(messages.finding_java_in_java_or_parasoft_tool_install_dir, installDir));
+        const javaFileName = os.platform() == "win32" ? "java.exe" : "java";
+        const javaPaths = [
+            "bin", // Java installation
+            "bin/dottest/Jre_x64/bin", // dotTEST installation
+            "bin/jre/bin" // C/C++test or Jtest installation
+        ];
+
+        for (const path of javaPaths) {
+            const javaFilePath = pt.join(installDir, path, javaFileName);
+            if (fs.existsSync(javaFilePath)) {
+                return javaFilePath;
+            }
+        }
+
+        return undefined;
+    }
+
+    private async convertReportsWithJava(javaPath: string, sourcePaths: string[]) : Promise<RunDetails> {
+        core.debug(messages.using_java_to_convert_report);
+        const jarPath = pt.join(__dirname, "SaxonHE12-2J/saxon-he-12.2.jar");
+        const xslPath = pt.join(__dirname, "cobertura.xsl");
+        const coberturaReports: string[] = [];
+
+        for (const sourcePath of sourcePaths) {
+            core.info(messagesFormatter.format(messages.converting_coverage_report_to_cobertura, sourcePath));
+            const outPath = sourcePath.substring(0, sourcePath.lastIndexOf('.xml')) + '-cobertura.xml';
+
+            const commandLine = `"${javaPath}" -jar "${jarPath}" -s:"${sourcePath}" -xsl:"${xslPath}" -o:"${outPath}" -versionmsg:off`;
```

Comment:
The parameter `pipelineBuildWorkingDirectory` needs to be specified. Otherwise, the `filename` in the generated Cobertura report will contain `${platform_working_directory}`, which will require additional processing to remove later.